jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -250,6 +266,15 @@ public boolean isRequestBufferEmpty()
         return _retainableByteBuffer == null || _retainableByteBuffer.isEmpty();
     }
 
+    @Override
+    public void fillInterested()
+    {
+        if (_readCallback != null)
+            getEndPoint().fillInterested(_readCallback);
+        else
+            super.fillInterested();
+    }
```

Comment:
This feels asymmetric and another "if" in critical path.     Would be better to make the _readCallback in AbstractConnection extensible rather than duplicated.